laiteluettelo.php

```diff
@@ -85,126 +80,254 @@
+
+  if (in_array("SP-FSNN-CO-NOC", $valitut_sarakkeet)) {
+    $worksheet->write($excelrivi, $excelsarake++, t("NOC e/kk"), $format_bold);
+  }
+  if (in_array("SP-FSNN-CO-MDM", $valitut_sarakkeet)) {
+    $worksheet->write($excelrivi, $excelsarake++, t("MDM e/kk"), $format_bold);
+  }
+  if (in_array("SP-FSNN-CO-MAINTENANCE", $valitut_sarakkeet)) {
+    $worksheet->write($excelrivi, $excelsarake++, t("OnSite Preventive Maintenance"), $format_bold);
+  }
+  if (in_array("SP-FSNN-SE-LCC", $valitut_sarakkeet) or in_array("SP-FSNN-SE-LCM", $valitut_sarakkeet)) { 
+    $worksheet->write($excelrivi, $excelsarake++, t("LCC/LCM e/kk"), $format_bold);
+  }
+  $worksheet->write($excelrivi, $excelsarake++, t("LCC/LCM SLA"), $format_bold);
+  $worksheet->write($excelrivi, $excelsarake++, t("LCC P��ttymisp�iv�"), $format_bold);
+
+  $excelrivi++;
+  $excelsarake = 0;
+
+  // Kirjoitetaan laiterivit
+  while ($row = mysql_fetch_assoc($laiteresult)) {
 
-    // Laite-/palvelukohtaiset valinnaiset kent�t
-
-    // Laitetunnus
-    // Sarjanumero
-    // Tuotenumero
-    // Tuotemerkki
-    // Valmistaja
-    // SLA
-    // VC numero / End date
-
-    // Nimitys(palvelu)
-    // Palvelu alku/loppu
-    // Kplhinta
-
-    foreach ($row as $key => $value) {
-      if (in_array($key, $valitut_sarakkeet) and !in_array($key, $sopimuskohtaiset)) {
-
-        if (is_numeric($value) and $key == "hinta") {
-          $value = str_replace(".", ",", hintapyoristys($value))." ".t("e / kk");
-          // Jos laitteiden kappalem��r� ei vaikuta palvelun hintaan
-          if ($row['netto'] != '') {
-            $value .= " **";
-            $lisainffot = "".t("** Palvelukohtainen hinta");
-          }
-        }
-
-        if ($eka_laiterivi) {
-          // Valittujen laiterivien headerit
-          $worksheet->write($excelrivi, $excelsarake, t("{$key}"),     $format_bold);
-          $worksheet->write($excelrivi+1, $excelsarake++, $value);
-        }
-        else {
-          $worksheet->write($excelrivi, $excelsarake++, $value);
-        }
+    // Haetaan laitekohtaiset tiedot
+    $laitequery = "SELECT
+                   laite.*,
+                   avainsana.selitetark valmistaja,
+                   tuote.tuotemerkki malli
+                   FROM laite
+                   LEFT JOIN tuote on tuote.yhtio = laite.yhtio
+                     AND tuote.tuoteno    = laite.tuoteno
+                   LEFT JOIN avainsana on avainsana.yhtio = tuote.yhtio
+                     AND avainsana.laji   = 'TRY'
+                     AND avainsana.selite = tuote.try
+                   WHERE laite.yhtio      = '{$kukarow['yhtio']}'
+                   AND laite.tunnus = '{$row['laitetunnus']}'";
 
+    $laiteres = pupe_query($laitequery);
+
+    $laiterow = mysql_fetch_assoc($laiteres);
+
+    // Haetaan laitteen palvelut joiden hinnat ovat laitteiden lukum��r�st� riippuvaisia
+    $palveluquery = "SELECT nimitys, hinta
```

Comment:
@mikamyllynen Tässä näytäisi olevan aikamoinen määrä duplikaattikoodia. Ajaisiko tilausrivi.tuoteno in ('SITÄ','TÄTÄ','TOTA') saman asian?